dopamine/discrete_domains/run_experiment.py

```diff
@@ -468,6 +469,13 @@ def _checkpoint_experiment(self, iteration):
       experiment_data['logs'] = self._logger.data
       self._checkpointer.save_checkpoint(iteration, experiment_data)
 
+  def _run_experiment_loop(self):
+    """Runs training / evaluation loop."""
```

Comment:
The name and comment are a little hard for me to parse.  I think maybe overriding `run_experiment` (and possibly in that calling a method that checks the iteration number, if you're really concerned with repeating code) reads a little more clearly.

Reply:
Updated the docstring.

I tried to move the check into a separate function but this actually makes things more complicated (because the loop shouldn't run in some conditions) - so I tend to prefer the current implementation.

Reply:
I don't understand the reasoning.  `run_experiment` makes sense to me, semantically, where `run_experiment_loop` is hard for me to understand as a thing.  If the loop part of it shouldn't be run in every case, then early exit as was done in the implementation before this patch inside the method and after this patch in a parent method, can still be done.

Is it the check that sometimes shouldn't be run?

Reply:
This is still outstanding.

Reply:
right but the 5 lines of code that run the check and logging then would have to be copied pasted. I wanted to avoid that.
If it's a naming issue I can change the name to `_ run_iterations`.

Reply:
I do think that `_run_iterations` is much better than `_run_experiment_loop`.

As for choosing to create `run_iterations` in the first place, I think it's ok to call it a minor difference of opinion.  Especially with the new name, it's not going to make a whole lot of difference.

LGTM

Reply:
Done.

Renamed to `_run_iterations`.